tests/foreman/cli/test_classparameters.py

```diff
@@ -282,38 +439,24 @@ def test_positive_validate_default_value_required_check(self):
         4.  Submit the change.
 
         @assert: No error raised for non-empty default value
-
-        @caseautomation: notautomated
-
-        @CaseLevel: System
-        """
-
-    @run_only_on('sat')
-    @stubbed()
-    @tier3
-    def test_negative_validate_matcher_value_required_check(self):
-        """Error raised for blank matcher Value - Required check.
-
-        @id: 5f9a65d7-b4e4-42d1-8e29-220f41e7450f
-
-        @steps:
-
-        1.  Override the parameter.
-        2.  Create a matcher for Parameter for some attribute.
-        3.  Dont provide Value for matcher. Keep blank.
-        4.  Set '--required' check.
-        5.  Submit the change.
-
-        @assert: Error raised for blank matcher value.
-
-        @caseautomation: notautomated
-
-        @CaseLevel: System
         """
+        sc_param_id = self.sc_params_ids_list.pop()
+        SmartClassParameter.update({
+            'parameter-type': 'boolean',
```

Comment:
BTW, 'required' checkbox has nothing in common with SC params itself, but more with 'hosts'. Check that bug out for more details:
https://bugzilla.redhat.com/show_bug.cgi?id=1352491
Just to make sure that you understand logic in a right way.
Even in UI, we need to override parameter on the host page to see how required flag is working